GAME/TOMB4FX.C

```diff
@@ -355,14 +389,162 @@ void TriggerBlood(int x, int y, int z, int a4, int num)
 
 void TriggerExplosionBubble(int x, int y, int z, short room_num)
 {
-	S_Warn("[TriggerExplosionBubble] - Unimplemented!\n");
+	int dx = lara_item->pos.x_pos - x;
+	int dz = lara_item->pos.z_pos - z;
+	if (dx >= -16384 && dx <= 16384 && dz >= -16384 && dz <= 16384)
+	{
+		struct SPARKS* sptr = &spark[GetFreeSpark()];
+		sptr->sR = 128;
+		sptr->dR = 128;
+		sptr->dG = 128;
+		sptr->dB = 128;
+		sptr->On = 1;
+		sptr->Life = 24;
+		sptr->sLife = 24;
+		sptr->sG = 64;
+		sptr->sB = 0;
+		sptr->ColFadeSpeed = 8;
+		sptr->FadeToBlack = 12;
+		sptr->TransType = 2;
+		sptr->x = x;
+		sptr->y = y;
+		sptr->z = z;
+		sptr->Xvel = 0;
+		sptr->Yvel = 0;
+		sptr->Zvel = 0;
+		sptr->Friction = 0;
+		sptr->Flags = 2058;
+		sptr->Scalar = 3;
+		sptr->Gravity = 0;
+		sptr->Def = objects[DEFAULT_SPRITES].mesh_index + 13;
+		sptr->MaxYvel = 0;
+		int size = (GetRandomControl() & 7) + 63;
```

Comment:
All variables must be declared at the start of a method.

Reply:
Done